internal/server/mcp_server.go

```diff
@@ -108,14 +134,47 @@ func handleToolsRegistration(
 			return nil, nil, fmt.Errorf("failed to read API spec from %s: %w", path, err)
 		}
 
-		// Overwrite the Trento URL in the OpenAPI
-		if len(oasDoc.Servers) > 0 {
-			oasDoc.Servers[0].URL = serveOpts.TrentoURL
+		// Determine the server base URL for this spec:
+		// 1. If no TrentoURL is passed, rely on the oas path,
+		// if is remote (http/https), replace the server with a derived baseURL (scheme://host[:port]).
+		if serveOpts.TrentoURL == "" && (strings.HasPrefix(path, "http://") || strings.HasPrefix(path, "https://")) {
+			newAPIServerURL := ""
+
+			parsedURL, err := url.Parse(path)
+
+			if err == nil && parsedURL.Scheme != "" && parsedURL.Host != "" {
+				newAPIServerURL = fmt.Sprintf("%s://%s", parsedURL.Scheme, parsedURL.Host)
+			}
+
+			// If there are already a server, replace it.
+			if len(oasDoc.Servers) > 0 {
+				slog.DebugContext(ctx, "replacing server URL in OpenAPI spec",
+					"path", path,
+					"old_url", oasDoc.Servers[0].URL,
+					"current_url", newAPIServerURL,
+				)
+
+				oasDoc.Servers[0].URL = newAPIServerURL
+				// 2. If TrentoURL is explicitly set (non-empty), leave servers untouched.
```

Comment:
question:
I think this comment is wrong.
It says `If TrentoURL is explicitly set` but you enter here if `if serverOpts.TrentoURL == ""`, which are opposite things.
This `else` statement simply creates the list if it came empty, right?